std/experimental/color/rgb.d

```diff
@@ -0,0 +1,446 @@
+}
+
+
+T linearTosRGB(T)(T s) if(isFloatingPoint!T)
+{
+    if(s <= T(0.0031308))
+        return T(12.92) * s;
+    else
+        return T(1.055) * s^^T(1.0/2.4) - T(0.055);
+}
+T sRGBToLinear(T)(T s) if(isFloatingPoint!T)
+{
+    if(s <= T(0.04045))
+        return s / T(12.92);
+    else
+        return ((s + T(0.055)) / T(1.055))^^T(2.4);
+}
+
+T linearToGamma(T, T gamma)(T v) if(isFloatingPoint!T)
+{
+    return v^^T(1.0/gamma);
+}
+T gammaToLinear(T, T gamma)(T v) if(isFloatingPoint!T)
+{
+    return v^^T(gamma);
+}
+
+T toGrayscale(bool linear, RGBColorSpace colorSpace = RGBColorSpace.sRGB, T)(T r, T g, T b) pure if(isFloatingPoint!T)
+{
+    // calculate the luminance (Y) value by multiplying the Y row of the XYZ matrix with the color
+    enum YAxis = RGBColorSpaceMatrix!(colorSpace, T)[1];
+
+    static if(linear)
+    {
+        return YAxis[0]*r + YAxis[1]*g + YAxis[2]*b;
+    }
+    else
+    {
+        // precise; convert to linear, then convert
+        return toGamma!colorSpace(YAxis[0]*toLinear!colorSpace(r) + YAxis[1]*toLinear!colorSpace(g) + YAxis[2]*toLinear!colorSpace(b));
+
+        // fast (standardised) approximations, performed in sRGB gamma space
+//        return T(0.299)*r + T(0.587)*g + T(0.114)*b; // Y'UV (PAL/NSTC/SECAM)
+//        return T(0.2126)*r + T(0.7152)*g + T(0.0722)*b; // HDTV
+    }
+}
+T toGrayscale(bool linear, RGBColorSpace colorSpace = RGBColorSpace.sRGB, T)(T r, T g, T b) pure if(isIntegral!T)
+{
+    import std.experimental.color.conv: convertPixelType;
+    alias F = FloatTypeFor!T;
+    return convertPixelType!T(toGrayscale!(linear, colorSpace)(convertPixelType!F(r), convertPixelType!F(g), convertPixelType!F(b)));
+}
+
+
+// helpers to parse color components from color component string
+template canFind(string s, char c)
+{
+    static if(s.length == 0)
+        enum canFind = false;
+    else
+        enum canFind = s[0] == c || canFind!(s[1..$], c);
+}
+template allIn(string s, string chars)
+{
+    static if(chars.length == 0)
+        enum allIn = true;
+    else
+        enum allIn = canFind!(s, chars[0]) && allIn!(s, chars[1..$]);
+}
+template anyIn(string s, string chars)
+{
+    static if(chars.length == 0)
+        enum anyIn = false;
+    else
+        enum anyIn = canFind!(s, chars[0]) || anyIn!(s, chars[1..$]);
+}
+template notIn(string s, string chars)
+{
+    static if(chars.length == 0)
+        enum notIn = char(0);
+    else static if(!canFind!(s, chars[0]))
+        enum notIn = chars[0];
+    else
+        enum notIn = notIn!(s, chars[1..$]);
+}
+
+unittest
+{
+    static assert(canFind!("string", 'i'));
+    static assert(!canFind!("string", 'x'));
+    static assert(allIn!("string", "sgi"));
+    static assert(!allIn!("string", "sgix"));
+    static assert(anyIn!("string", "sx"));
+    static assert(!anyIn!("string", "x"));
+}
+
+
+// 3d linear algebra functions (this would ideally live somewhere else...)
```

Comment:
That's a whole new can of worms...
There needs to be a linear algebra module, but until there is, I can't think where else to put this. I don't think it belongs in std.math, and that seems like a much more invasive PR.